enumerable.rb

```diff
@@ -0,0 +1,129 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    items = is_a?(Range) ? to_a : self
+    index = 0
+    while index < items.length
+      yield(items[index])
+      index += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return enum_for unless block_given?
+
+    items = is_a?(Range) ? to_a : self
+    index = 0
+    while index < items.length
+      yield(items[index], index)
+      index += 1
+    end
+    self
+  end
+
+  def my_select
+    return enum_for unless block_given?
+
+    array = []
+    my_each { |item| array << item if yield(item) }
+    array
+  end
+
+  def my_all?(*args)
+    unless args[0].nil?
+      my_each { |x| return false if (x =~ args[0]).nil? }
+      return true
+    end
+    unless block_given?
+      my_each { |x| return false if x.nil? || x == false }
+      return true
+    end
+    my_each { |x| return false unless yield(x) }
+    true
+  end
+
+  def my_any?(*args)
+    unless args[0].nil?
+      my_each { |x| return true if x =~ args[0] }
+      return false
+    end
+    unless block_given?
+      my_each { |x| return true if !x.nil? && (x != false) }
+      return false
+    end
+    my_each { |x| return true if yield(x) }
+    false
+  end
+
+  def my_none?(*args)
+    unless args[0].nil?
+      my_each { |x| return false if x =~ args[0] }
+      return true
+    end
+    unless block_given?
+      my_each { |x| return false unless x.nil? || (x == false) }
+      return true
+    end
+    my_each do |x|
+      return false if yield(x)
+    end
+    true
+  end
```

Comment:
You are not considering the option of giving a class or a number as an argument. For example:
`some_enumerable.my_none?(String)`
`some_enumerable.my_none?(6)`
Right now, those examples return false on truthy cases.

Reply:
I improved my_none method to support Class and object as argument